libdrgn/kdump.c

```diff
@@ -139,3 +141,65 @@ struct drgn_error *drgn_program_set_kdump(struct drgn_program *prog)
 	kdump_free(ctx);
 	return err;
 }
+
+struct drgn_error *drgn_program_cache_prstatus_kdump(struct drgn_program *prog)
+{
+	kdump_num_t ncpus;
+	kdump_status ks;
+
+	ks = kdump_get_number_attr(prog->kdump_ctx, "cpu.number", &ncpus);
+	if (ks != KDUMP_OK)
+		return drgn_error_format(DRGN_ERROR_OTHER,
+					 "kdump_get_number_attr(cpu.number): %s",
+					 kdump_get_err(prog->kdump_ctx));
+
+	/*
+	 * Note that in the following loop we never call kdump_attr_unref()
+	 * on prstatus_ref, nor kdump_blob_unpin() on the prstatus blob that
+	 * get from libkdumpfile. Since drgn is completely read-only as a
+	 * consumer of that library, we "leak" both the attribute reference
+	 * and blob pin until kdump_free() is called which will clean up
+	 * everything for us.
+	 */
+	for (int i = 0; i < ncpus; i++) {
```

Comment:
Since our loop bound is a `kdump_num_t`, we probably want `i` to be `kdump_num_t`, too.